### PR TITLE
Docstring fix data converter

### PR DESCRIPTION
**Pull request by feipenghe**
This PR partially corrects #610. 

* It will fix all scripts under `/forte/forte/data/converter`
    * `forte/data/converter/converter.py`
    * `forte/data/converter/feature.py`

### Description of changes
Describe what are the changes, and how they solve the problem.

### Possible influences of this PR.
It fixes docstrings of data converter to have the following features
* valid hyperlinks.
* docstring in major functions are shown correctly
* classes in `data.rst` includes all classes that needs documentation.
* correct indentation.

### Test Conducted
Check process
1. go through classes in the corresponding folders and check if it exists in sphinx documentation
2. use automation tool to check invalid hyperlinks
3. check indentation issues by visually going through sphinx documentation. 

### What's fixed?
* `LinkExtractor` and `SubwordExtractor` wasn't in sphinx configuration file. There were also some indentation issues in `default_configs()`.  Screenshots below shows effects after fixing.
image
image

* Two `Note` weren't aligned on a same indentation level initially. The screenshot below shows effects after fixing.
image

* Fixed hyperlink of `BaseExtractor` and indentation for `dim`, `dtype`. Screenshot below shows effects after fixing.

image

### CODE REVIEW

**Comment by hunterhector** [comment]
Don't use the word "fixes" in PR, otherwise, it will close the PR. https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

File: ft/onto/base_ontology.py [hunk 1]
```diff
@@ -476,6 +477,16 @@ def __init__(self, pack: DataPack, begin: int, end: int):
         super().__init__(pack, begin, end)
 
 
+@dataclass
+class Body(Annotation):
```
Comment (hunterhector) [inline ft/onto/base_ontology.py:481]: These does not belong to this PR, remove them

**Review by hunterhector — COMMENTED** [review]
For docstring fixing PRs, attach the screenshot to help with the review. 

In the PR message, say specifically which files are fixed, not use the general description

File: ft/onto/base_ontology.py (hunk 1 above)
Comment (feipenghe) [reply ft/onto/base_ontology.py]: > Don't use the word "fixes" in PR, otherwise, it will close the PR. https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

I think you mentioned something like `partially fix` in the meeting. I checked the link it seems like all keywords would close the PR. Should I avoid using the keywords like `partially correct`?

Comment (hunterhector) [reply ft/onto/base_ontology.py]: You can use any verbs outside the list:
close
closes
closed
fix
fixes
fixed
resolve
resolves
resolved

**Comment by hunterhector** [comment]
Should also run the CI checks locally for the docs before pushing it, waiting the CI to run online will take a lot of time.

**Comment by feipenghe** [comment]
Fixed hyperlink of `BaseExtractor` and indentation for `dim`, `dtype`. Screenshot below shows effects after fixing.



image

**Review by hunterhector — APPROVED** [review]
